images/decode.go

```diff
@@ -11,13 +11,35 @@ import (
 	"io/ioutil"
 	"net/http"
 	"os"
+	"regexp"
 	"time"
+	"unicode/utf8"
 
 	"golang.org/x/image/webp"
 
 	"github.com/ethereum/go-ethereum/log"
 )
 
+var (
+	htmlCommentRegex = regexp.MustCompile(`(?i)<!--([\\s\\S]*?)-->`)
+	svgRegex         = regexp.MustCompile(`(?i)^\s*(?:<\?xml[^>]*>\s*)?(?:<!doctype svg[^>]*>\s*)?<svg[^>]*>[^*]*<\/svg>\s*$`)
+)
+
+// Is returns true if the given buffer is a valid SVG image.
+func IsSVG(buf []byte) bool {
+	var isBinary bool
+	if len(buf) < 24 {
+		isBinary = false
+	}
+	for i := 0; i < 14; i++ {
+		charCode, _ := utf8.DecodeRuneInString(string(buf[i]))
+		if charCode == 65533 || charCode <= 8 {
+			isBinary = true
+		}
+	}
+	return !isBinary && svgRegex.Match(htmlCommentRegex.ReplaceAll(buf, []byte{}))
```

Comment:
Once you cover this function with a unit test, it would be a good idea to use non-trivial SVG(s) to see if the complicated regex `svgRegex` is actually good enough. I don't know if any LLM recommended that to you, but I had chatgpt sharing blatantly wrong regexes to me up to a point I gave up and wrote by hand.

Reply:
Also to be honest, I don't really know how non-trivial SVG(s) can be
Since it's markup it's so hard to have multiple ways to do SVGs.
If you have any examples that'd help me a lot.